### PR TITLE
Fix New-Item -Path with wildcard char

### PR DESCRIPTION
**Pull request by kwkam**
## PR Summary

Unescape non-literal path when -Name is not set.
This works around for New-Item treating -Path as literal path while
it can also be globbable. For example, assuming there is ```"[file]1"```
in current directory, and tab completion suggests ```'./`[file`]1'```
for the -Path argument, but
```pwsh
New-Item -Path './`[file`]2'
```
will create a file named ```"`[file`]2"``` instead of ```"[file]2"```.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by kwkam** [comment]
Test result depends on WildcardPattern::Escape in #7407

File: test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1 [hunk 1]
```diff
@@ -246,6 +259,21 @@ Describe "New-Item with links" -Tags @('CI', 'RequireAdminOnWindows') {
...
     }
 
+    It "Should create symbolic link with name contains special char" {
+        $null = New-Item -Path $tmpDirectory -Name $testlinkSrcSpName -ItemType File
+        $FullyQualifiedLSrcSp | Should -Exist
+
+        $null = New-Item -Path $FullyQualifiedLinkSp -Target $FullyQualifiedLSrcSp -ItemType SymbolicLink
+        $FullyQualifiedLinkSp | Should -Exist
+
+        $expectedTarget = Join-Path $tmpDirectory $testlinkSrcSpName
```
Comment (kvprasoon) [inline test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1]: Use named parameters of `Join-Path` cmdlet. (`-Path` and `-ChildPath`)

File: test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1 [hunk 2]
```diff
@@ -246,6 +259,21 @@ Describe "New-Item with links" -Tags @('CI', 'RequireAdminOnWindows') {
...
+
+        $fileInfo = Get-ChildItem $FullyQualifiedLinkSp
```
Comment (kvprasoon) [inline test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1]: Use named parameter of `Get-ChildItem` cmdlet. (`-Path`)

File: test/powershell/Modules/Microsoft.PowerShell.Management/Remove-Item.Tests.ps1 [hunk 3]
```diff
@@ -3,7 +3,10 @@
 Describe "Remove-Item" -Tags "CI" {
     $testpath = $TestDrive
     $testfile = "testfile.txt"
+    $testfileSpName = "[testfile].txt"
```
Comment (kvprasoon) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Remove-Item.Tests.ps1]: Align variable definitions.
```
$testpath       = $TestDrive 
$testfile       = "testfile.txt" 
$testfile       = "testfile.txt" 
$testfileSpName = "[testfile].txt" 
$testfileSp     = "``[testfile``].txt" 
$testfilepath   = Join-Path -Path $testpath -ChildPath $testfile 
$testfilepath   = Join-Path -Path $testpath -ChildPath $testfile 
$testfilepathSp = Join-Path -Path $testpath -ChildPath $testfileSp 
```

**Review by kvprasoon — CHANGES_REQUESTED** [review]

**Pull request closed without merging**